lua/runner/init.lua

```diff
@@ -28,29 +28,44 @@ M.set_handler = function(filetype, handler)
   M._handlers[filetype] = handler
 end
 
+--- @param bufnr integer|nil
 M.run = function(bufnr)
-  local buffer
   if bufnr == nil or bufnr == 0 then
-    buffer = vim.api.nvim_get_current_buf()
-  else
-    buffer = bufnr
+    bufnr = vim.api.nvim_get_current_buf()
   end
 
-  if buffer == utils._terminal_buffer then
+  if bufnr == utils._terminal_buffer then
     helpers.shell_handler(utils._last_command, false)()
     return
   end
 
-  local filetype = vim.filetype.match { buf = buffer }
+  local filetype = vim.filetype.match { buf = bufnr }
 
   local handler = M._handlers[filetype]
 
   if not handler then
-    print(string.format("No handler defined for filetype '%s'", filetype))
+    print('No handler defined for filetype ' .. filetype)
     return
   end
 
-  handler(buffer)
+  handler(bufnr)
+end
+
+--- @param bufnr integer|nil
+M.autorun = function(bufnr)
+  M.run(bufnr)
+  vim.api.nvim_create_autocmd('BufWritePost', {
+    group = vim.api.nvim_create_augroup('AutoRunner', { clear = true }),
+    pattern = '*',
+    callback = function()
+      helpers.shell_handler(utils._last_command, false)()
```

Comment:
You're restricting the command to only be a shell command. That way, if for example you're running a lua file, which uses the command handler to run `luafile %`, it breaks. Please update it to run the last handler regardless of which type it is.
You can add a `_last_handler` function somewhere that you overwrite each time you run the file, that just encapsulates the handler being run. When `autorun` runs, it just calls that function.